library/waku_thread/waku_thread.nim

```diff
@@ -83,20 +90,24 @@ proc createWakuThread*(): Result[void, string] =
 proc stopWakuNodeThread*() =
   running.store(false)
   joinThread(ctx.thread)
-
-  ctx.reqChannel.close()
-  ctx.respChannel.close()
-
   freeShared(ctx)
 
-proc sendRequestToWakuThread*(req: InterThreadRequest): Result[string, string] =
+proc sendRequestToWakuThread*(reqType: RequestType,
+                              reqContent: pointer): Result[string, string] =
 
-  ctx.reqChannel.send(req)
+  let req = InterThreadRequest.createShared(reqType, reqContent)
 
-  var resp = ctx.respChannel.tryRecv()
-  while resp[0] == false:
-    resp = ctx.respChannel.tryRecv()
-    os.sleep(1)
+  ## Sending the request
+  let sentOk = ctx.reqChannel.trySend(req)
```

Comment:
can it happen that requests are answered out-of-order because of `async`?

Reply:
> can it happen that requests are answered out-of-order because of `async`?

Good point! I believe this couldn't happen as requests are attended sequentially, thanks to the `waitFor` in https://github.com/waku-org/nwaku/blob/01793d2156bf39162995459a59777a36dc4e0b29/library/waku_thread/waku_thread.nim#L58

On the other hand, once we will apply the "signal" approach we'll have a better synchronization between both threads.